### PR TITLE
EVG-16501: clean up error message style

### PR DESCRIPTION
**Pull request by Kimchelly**
Jira: https://jira.mongodb.org/browse/EVG-16501

Clean up error message style.

### CODE REVIEW

File: build_options.go [hunk 1]
```diff
@@ -35,24 +34,11 @@ func (o BuildOptions) GetBuildInfo(version string) BuildInfo {
...
-
-	if o.Edition == "" {
-		errs = append(errs, "edition definition is missing")
-	}
-
-	if len(errs) != 0 {
-		return errors.Errorf("%d errors: [%s]",
-			len(errs), strings.Join(errs, "; "))
-	}
+	catcher.NewWhen(o.Target == "", "missing target")
```
Comment (julianedwards) [inline build_options.go:39]: `must specify a target`
`must specify an architecture`
`must specify an edition`

File: feed.go [hunk 2]
```diff
@@ -244,7 +244,7 @@ func (feed *ArtifactsFeed) GetArchives(releases []string, options BuildOptions)
 
 			version, ok := feed.GetVersion(rel)
 			if !ok {
-				catcher.Add(errors.Errorf("no version defined for %s", rel))
+				catcher.Add(errors.Errorf("no version defined for release '%s'", rel))
```
Comment (julianedwards) [inline feed.go:247]: `catcher.Errorf`

File: fetch.go [hunk 3]
```diff
@@ -95,19 +95,19 @@ func DownloadFile(ctx context.Context, url, fileName string) error {
...
 	if err != nil {
-		return errors.Wrap(err, "problem downloading file")
+		return errors.Wrap(err, "downloading file")
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode >= 300 {
 		grip.Warning(os.Remove(fileName))
-		return errors.Errorf("encountered error %d (%s) for %s", resp.StatusCode, resp.Status, url)
+		return errors.Errorf("received status code %d (%s) for %s", resp.StatusCode, resp.Status, url)
```
Comment (julianedwards) [inline fetch.go:104]: `'%s'`

File: versions.go [hunk 4]
```diff
@@ -188,7 +188,7 @@ func createNewMongoDBVersion(parsedVersion LegacyMongoDBVersion) (*NewMongoDBVer
 		if len(split) > 0 && len(split[0]) > len(devReleaseTag) {
 			v.devReleaseNumber, err = strconv.Atoi(split[0][len(devReleaseTag):])
 			if err != nil {
-				return nil, errors.Wrapf(err, "couldn't parse development release number")
+				return nil, errors.Wrapf(err, "parsing development release number")
```
Comment (julianedwards) [inline versions.go:191]: `errors.Wrap`

File: versions.go [hunk 5]
```diff
@@ -235,7 +235,7 @@ func createLegacyMongoDBVersion(version string) (*LegacyMongoDBVersion, error) {
 
 			v.rcNumber, err = strconv.Atoi(rcPart[0][2:])
 			if err != nil {
-				return nil, errors.Wrapf(err, "couldn't parse release candidate number")
+				return nil, errors.Wrapf(err, "parsing release candidate number")
```
Comment (julianedwards) [inline versions.go:238]: `errors.Wrap`

**Review by julianedwards — CHANGES_REQUESTED** [review]

File: build_options.go (hunk 1 above)
Comment (Kimchelly) [reply build_options.go]: Done.

File: feed.go (hunk 2 above)
Comment (Kimchelly) [reply feed.go]: Fixed.

File: fetch.go (hunk 3 above)
Comment (Kimchelly) [reply fetch.go]: Fixed.

File: versions.go (hunk 4 above)
Comment (Kimchelly) [reply versions.go]: Fixed.

File: versions.go (hunk 5 above)
Comment (Kimchelly) [reply versions.go]: Fixed.

**Review by julianedwards — APPROVED** [review]

**Pull request merged**